services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -309,12 +305,15 @@ class HankeController(
         }
     }
 
-    private fun Hanke.validateUpdatable(updatedHanke: Hanke, hankeTunnusFromPath: String) {
-        if (hankeTunnusFromPath != updatedHanke.hankeTunnus) {
-            throw HankeArgumentException("Hanketunnus not given or doesn't match the hanke data")
-        }
-        if (perustaja != null && perustaja != updatedHanke.perustaja) {
-            throw HankeArgumentException("Updating perustaja not allowed.")
+    private fun validateUpdatable(existing: Hanke, updated: Hanke, hankeTunnusFromPath: String) {
+        val tunnusMatch =
+            listOf(existing.hankeTunnus, updated.hankeTunnus).all { it == hankeTunnusFromPath }
```

Comment:
Since the existing hanke is loaded with the hanketunnus from the path, they can't be different. So there's no reason to check that they're the same.

Reply:
Fixed.